Source/dllmain.cpp

```diff
@@ -0,0 +1,169 @@
+#pragma once
+#define _WINSOCKAPI_
+#define WIN32_LEAN_AND_MEAN
+#include <winsock2.h>
+#include <windows.h>
+#include <fstream>
+#include <comdef.h>
+#include <string>
+#include <sstream>
+#include <thread>
+#include <mutex>   
+#include <condition_variable>
+#include "TlHelp32.h"
+#pragma comment (lib, "Ws2_32.lib")
+#pragma warning(disable : 4996)
+
+HHOOK g_hHook = NULL;
+HINSTANCE g_hDll = NULL;
+#define PORT 17078
+
+BOOL APIENTRY DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved);
+extern "C" __declspec(dllexport) BOOL SetMsgHook(char* path);    // Exported
+extern "C" __declspec(dllexport) BOOL UnsetMsgHook();
+extern "C" __declspec(dllexport) BOOL OpenLogFile(char* path);
+LRESULT CALLBACK SysMsgProc(int nCode, WPARAM wParam, LPARAM lParam);
+
+SOCKET sock;
+char buff[sizeof(MSG)];
+struct sockaddr_in Sender_addr;
+bool socketInitialized = false;
+std::ofstream myfile;
+
+extern "C" __declspec(dllexport) HINSTANCE getDllHinstance() {
+    return g_hDll;
+}
+
+extern "C" __declspec(dllexport) int getSocketPort() {
+    return PORT;
+}
+
+void printLog(const std::string& message)
+{
+    if (myfile.is_open()) {
+        myfile << message << std::endl;
+    }
+}
+
+void messageToFile(LPMSG aMsg) {
+    if (myfile.is_open()) {
+        std::stringstream ss;
+        ss << aMsg->time << ";";
+        ss << aMsg->message << ";";
+        ss << aMsg->hwnd << ";";
+        ss << aMsg->lParam << ";";
+        ss << aMsg->wParam << ";";
+        ss << aMsg->pt.x << ";";
+        ss << aMsg->pt.y << ";";
+        printLog(ss.str());
+    }
+}
+
+BOOL APIENTRY DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+    switch (fdwReason)
+    {
+    case DLL_PROCESS_ATTACH:
+        g_hDll = hinstDLL;
+        //MessageBoxA(0, "Attach now", "Attach now", 0);
+        return true;
+    case DLL_PROCESS_DETACH:
+        UnsetMsgHook();
+        return true;
+    }
+}
+
+BOOL OpenLogFile(char* path) {
+    if (path)
+        myfile.open(path);
+    return myfile.is_open();
+}
+
+void infinityWait() {
+    std::condition_variable cv;
+    std::mutex m;
+    std::unique_lock<std::mutex> lock(m);
+    cv.wait(lock, [] {return false; });
+}
+
+void initSocket() {
+    WSADATA wsaData;
+    WSAStartup(MAKEWORD(2, 2), &wsaData);
+    sock = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
`SOCK_STREAM` is for TCP socket. We need TCP to make sending the data reliable. This also can use self-assigned port when passing port number == 0. Returned socket should have first free port. This way we avoid the situation when port is already in use.